packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -132,29 +136,71 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
     @inject(MonacoTextModelService)
     protected readonly textModelService: MonacoTextModelService;
 
+    private async openWith(commandId: string, resource: URI, columnOrOptions?: ViewColumn | TextDocumentShowOptions, viewTypeId?: string): Promise<boolean> {
+        if (!resource) {
+            throw new Error(`${commandId} command requires at least URI argument.`);
+        }
+        if (!URI.isUri(resource)) {
+            throw new Error(`Invalid argument for ${commandId} command with URI argument. Found ${resource}`);
+        }
+
+        let options: TextDocumentShowOptions | undefined;
+        if (typeof columnOrOptions === 'number') {
+            options = {
+                viewColumn: columnOrOptions
+            };
+        } else if (columnOrOptions) {
+            options = {
+                ...columnOrOptions
+            };
+        }
+
+        const uri = new TheiaURI(resource);
+        const editorOptions = DocumentsMainImpl.toEditorOpenerOptions(this.shell, options);
+
+        let opener: OpenHandler | undefined;
+        if (viewTypeId && viewTypeId !== 'default') {
+            const lowerViewType = viewTypeId.toLowerCase();
+            // Theia set custom editor id using 'custom-editor-' prefix
+            const customViewType = `custom-editor-${lowerViewType}`;
```

Comment:
I feel a bit unfomfortable with this piece of code being here as it replicates the knowledge of how custom editor opener id's are constructed outside of the relevant code. We should move that logic to `custom-editor-opener`as a function or static method.
Also, as I read the VS Code doc, the you can only match custom editor view types with this command. For example, `openWith('html')` should not work because it's not a custom editor. Am I reading this wrong?

Reply:
I agree with your opinion.
To accommodate your comments, I will proceed with the appropriate modifications.